application/templates/alertmanagerconfig.yaml

```diff
@@ -0,0 +1,13 @@
+{{- if .Values.alertmanagerConfig -}}
+{{- if .Values.alertmanagerConfig.enabled -}}
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: AlertmanagerConfig
+metadata:
+  name: {{ template "application.name" . }}
+  namespace: {{ template "application.namespace" . }}
+  labels:
+{{ toYaml .Values.alertmanagerConfig.selectionLabels | indent 4 }}
+spec:
+{{ toYaml .Values.alertmanagerConfig.spec | indent 4 }}
```

Comment:
why are we taking whole spec as input?

Reply:
because we don't want to limit the user on configuration level. Whole spec acts as the configuration for alertmanager.

Reply:
yes but isn't it possible that you just take some key value pairs and provide rest of the template here in charts?

Reply:
we can discuss this over call as well; but we do follow this practice where one can provide full spec

Reply:
We can but that way we will be limiting spec for customer. 
If we want to do that then we need to discuss this

Reply:
@rasheedamir  not sure in which template we are following this practice. I dont see any template where we are taking whole spec as input

Reply:
fixed